File: src/retin/toolbox/document/Flow2D.cpp
/**
 * \file Flow2D.cpp
 * \author Romain Negrel
 * \version 4.0
 */

#include "retin/toolbox/document/Flow2D.h"

#include "retin/toolbox/document/serialization/Serializers.h" 

RETIN_SERIALIZABLE_OBJECT_DEFINITIONS(Flow2D);

namespace retin {

void	Flow2D::resize(int width_,int height_) {
    nWidth = width_;
    nHeight = height_;
    pFlowX.resize(nWidth*nHeight);
    pFlowY.resize(nWidth*nHeight);
    memset(pFlowX.get(), 0, nWidth*nHeight*sizeof(float));
    memset(pFlowY.get(), 0, nWidth*nHeight*sizeof(float));
}

}

File: src/retin/toolbox/document/Flow2D.java
/*
 * To change this template, choose Tools | Templates
 * and open the template in the editor.
 */
package retin.toolbox.document;

/**
 *
 * @author romanegr
 */
public class Flow2D {
    private int width,height;
    private float[] flowX;
    private float[] flowY;

    public	Flow2D ()
    {
            this.width = 0;
            this.height = 0;
    }
    public Flow2D (int width,int height)
    {
            this.width = width;
            this.height = height;
            this.flowX = new float[width*height];
            this.flowY = new float[width*height];
    }

    public int getWidth() { return width; }
    public int getHeight() { return height; }
    public float[] getDataX() { return flowX; }
    public float[] getDataY() { return flowY; }
    
    public void resize (int width,int height){
        if(this.width == width || this.height == height){
            this.width = width;
            this.height = height;
            this.flowX = new float[width*height];
            this.flowY = new float[width*height];
        }
    }

    public void setValueX (int i,int j,float val) 
    {
            flowX[i+j*width] = val;
    }

    public float getValueX (int i,int j) 
    {
            return flowX[i+j*width];
    }
    
    public void setValueY (int i,int j,float val) 
    {
            flowY[i+j*width] = val;
    }

    public float getValueY (int i,int j) 
    {
            return flowY[i+j*width];
    }

    public	Object 	convertTo (Class classObject) throws Exception
    {
            if (classObject.equals(getClass()))
                    return this;
            throw new Exception ("Ne sait pas convertir de "+getClass().getName()+" vers "+classObject.getName());
    }
}
